### DISCUSSION TITLE
IEEE 802.1AS (PTP) Implementation

### DISCUSSION THREAD

**Discussion by debnathr** [CoRE-RG/CoRE4INET#35] [Q&A] [answered]
Hello CoRE4INET Group,

I am working on ATS, AVB, and TAS simulations using OMNeT++. I looked into your AVB and TT traffic "static_network" simulation. 

I saw that your group implemented the IEEE 802.1 Qat and CBS. My question is did your group also implement the IEEE 802.1AS (PTP) for the time synchronization? If not, then how the PTP part is handled in the simulation? If yes, then is the PTP implementation based on the IEEE 802.1 Qat standard? 

I will really appreciate if you could clarify my doubt.

Thanks and Best Regards,
Rubi

**Comment by Musteblume** [comment, marked as answer]
Hi Rubi,

i am happy to help with clarification. There is a module called DummySync in CoRE4INET. This models the timing behavior of sync protocols without the implemenation of a specific protocol. DummySync is an example of a ISync interface implementation. A specific sync protocol implementation can be added there.

Best regards
Philipp

**Comment by debnathr** [comment]
Hi Philipp,

Thank you so much for your reply. It helped.

Thanks and Regards,
Rubi

**Comment by debnathr** [comment]
Hi Philipp,

I have another question. Did your group also take into consideration the BE traffic? If so, was the strict priority algorithm implemented for scheduling the BE traffic? 

Thanks and Regards,
Rubi

**Comment by Musteblume** [comment]
Hi Rubi,

there are different traffic shaping algorithms implemented in CoRE4INET. In general BE traffic is considered in all our models. For a default AS6802 node BE traffic is not allowed to be send while a TT send window is open or if it would interfere with it. For a default TSN node it is dependent on the   Q priorities and the Qbv configuration. 

Best regards
Philipp

**Comment by Musteblume** [comment]
Here is a link to past work regarding effects of BE traffic in real-time ethernet networks: https://core-researchgroup.de/bib/eigene/slksh-bhcan-15.pdf

**Comment by debnathr** [comment]
Hi Philipp,

Thank you for the explanation and for sending me the link to the paper. 

Thanks and Regards,
Rubi
